src/main/java/org/spongepowered/api/plugin/Dependency.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.plugin;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Represents a dependency for a {@link Plugin}.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({})
+public @interface Dependency {
+
+    /**
+     * The plugin ID of the dependency.
+     *
+     * @return The dependency plugin ID
+     * @see Plugin#id()
+     */
+    String id();
+
+    /**
+     * The required version of the dependency.
+     *
+     * @return The required version, or an empty string if unspecified
+     */
+    String version() default ""; // TODO: Maven version ranges
+
+    /**
+     * If this dependency is optional for the plugin to work. By default
+     * this is {@code false}.
+     *
+     * @return True if the dependency is optional for the plugin to work
```

Comment:
I could add this to the Javadocs, however for the annotation `optional` is more clear IMO